### ISSUE TITLE
3.0.0: error on custom functions

### ISSUE THREAD

**Issue by niksubramanian** [postgraphql/postgraphql#390]
First of all, great library, thanks for all of the great work.

There seems to be an error that started occurring in the 3.0.0 branch related to custom functions. 

How to reproduce this issue:
## 1. create the tables, functions and populate the database
```SQL
# A first table
create table public.foo (
    id serial not null primary key,
    name text
);

# A second table
create table public.bar (
    id serial not null primary key,
    created_at timestamp default now(),
    foo_id int not null references public.foo(id)
);

# A function that returns the last item of the second table
create function public.foo_last_bar(foo public.foo) returns bar as $$
    select *
    from bar
    where bar.foo_id = foo.id
    order by created_at desc
    limit 1
$$ language sql stable;

# Some data
insert into foo (name) values ('postgraphql');
insert into bar (foo_id) values (1);
insert into bar (foo_id) values (1);
```

## 2. Execute a query in GraphiQL:
```GQL
fooById(id: 1) {
    lastBar{id}
}
```

## 3.A: The 3.0.0 branches' output:
```
{
  "data": {
    "fooById": {
      "lastBar": null
    }
  },
  "errors": [
    {
      "message": "value.get is not a function",
      "locations": [
        {
          "line": 2,
          "column": 14
        }
      ],
      "path": [
        "fooById",
        "lastBar",
        "id"
      ]
    }
  ]
}
```

## 3.B: The 2.6.0 branches' output:
```
{
  "data": {
    "fooById": {
      "lastBar": {
        "id": 2
      }
    }
  }
}
```

Any idea what might be causing this?

**Comment by Trupal00p** [comment]
Thanks for reporting this, just ran into it today.  For others who stumble upon this issue, I'm currently working around it by returning a list of results with the "setof" modifier instead of a single row variable.

**Comment by EyMaddis** [comment]
It is crashing at this line:
https://github.com/calebmer/postgraphql/blob/ade728ed8f8e3ecdc5fdad7d770c67aa573578eb/src/postgres/inventory/type/PgClassType.ts#L59

The `value` is not a `Map` and therefore does not have a `.get()` function.
I changed it to a simple

```js 
         getValue: value => return value[fieldName]
```

I guess that the value *should* be a map, but fails somewhere in the stack.
Stacktrace: 

```
TypeError: value.get is not a function
    at Object.getValue (postgraphql/src/postgres/inventory/type/PgClassType.ts:60:26)
    at resolve (postgraphql/src/graphql/schema/collection/createCollectionGqlType.ts:53:70)
    at resolveOrError (postgraphql/node_modules/graphql/execution/execute.js:474:12)
    at resolveField (postgraphql/node_modules/graphql/execution/execute.js:460:16)
    at postgraphql/node_modules/graphql/execution/execute.js:274:18
    at Array.reduce (native)
    at executeFields (postgraphql/node_modules/graphql/execution/execute.js:271:42)
    at collectAndExecuteSubfields (postgraphql/node_modules/graphql/execution/execute.js:719:10)
    at completeObjectValue (postgraphql/node_modules/graphql/execution/execute.js:701:10)
    at completeValue (/Volumes/SMPL/postgraphql/node_modules/graphql/execution/execute.js:598:12)
```

Either it was never a Map or is just sometimes not a map (@calebmer ?), I am not sure. It's the first time I touched the PostGraphQL code and even TypeScript.

A quick and dirty workaround would be:

```
         getValue: value => {
            if(value instanceof Map) {
              return value.get(fieldName)
            } else {
              return value[fieldName];
            }
          }
```

**Comment by calebmer** [comment]
There’s a fix for this here: https://github.com/postgraphql/postgraphql/pull/409

Let me know if that works 👍

@EyMaddis the value should always be a map. The issue was in a recent refactor that I did I forgot to convert the database value (an object) into a map 😣

The fix is in the PR.

**Comment by calebmer** [comment]
Fixed in 3.1.0 🎉
